tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -97,20 +92,33 @@ spec:
           repository=$(jq -r '.repository' <<< $component)
           imageTags=$(jq '.tags' <<< $component)
 
-          arches=$(get-image-architectures "${containerImage}" | jq -s 'map(.platform.architecture)')
+          # oras has very limited support for selecting the right auth entry,
+          # so create a custom auth file with just one entry.
+          AUTH_FILE=$(mktemp)
+          select-oci-auth "${containerImage}" > "$AUTH_FILE"
+
           # Just read the first from the list of architectures
+          arch_json=$(get-image-architectures "${containerImage}")
+          arches=$(jq -s 'map(.platform.architecture)' <<< $arch_json)
+          oses=$(jq -s 'map(.platform.os)' <<< $arch_json)
+
+          os=$(jq -r '.[0]' <<< $oses)
           arch=$(jq -r '.[0]' <<< $arches)
           name=$(jq -r '.name' <<< $component)
           git_sha=$(jq -r '.source.git.revision' <<< $component) # this sets the value to "null" if it doesn't exist
+          media_type=$(skopeo inspect --raw "docker://${containerImage}" | jq -r .mediaType)
+          oras_args=()
+          if [[ "$media_type" == "application/vnd.docker.distribution.manifest.list.v2+json" ]]\
+            || [[ "$media_type" == "application/vnd.oci.image.index.v1+json" ]]; then
+            oras_args+=("--platform $os/$arch")
+          fi
 
-          origin_digest=$(skopeo inspect \
-            --override-arch "${arch}" \
-            --no-tags \
-            --format '{{.Digest}}' \
-            "docker://${containerImage}" 2>/dev/null)
+          origin_digest=$(oras resolve --registry-config "$AUTH_FILE" "${oras_args[@]}" "${containerImage}" 2>/dev/null)
 
-          RESULTS_JSON=$(jq --arg i "$i" --argjson arches "$arches" --arg name "$name" --arg sha "$origin_digest" \
-            '.images[$i|tonumber] += {"arches": $arches, "name": $name, "shasum": $sha, "urls": []}' <<< $RESULTS_JSON)
+          RESULTS_JSON=$(jq --arg i "$i" --argjson arches "$arches" --argjson oses "$oses" --arg name "$name" \
+            --arg sha "$origin_digest" \
```

Comment:
Would be a little nicer to have more content on this line before the `\` but will leave that up to you. Feel free to resolve without change

Reply:
you mean why I am including `oses`?

Reply:
No, I mean `--arg sha "$origin_digest"` is the only thing on the line when the line can be like 50 characters longer before going over yamllint

Reply:
I read context not content...

I feel like breaking up the next line would make it less readable. I vote to keep as-is.